CromwellRefdiskManifestCreator/src/main/java/org/broadinstitute/manifestcreator/CromwellRefdiskManifestCreatorApp.java

```diff
@@ -148,44 +149,52 @@ private static long calculateCrc32c(File file) throws CRC32CException {
   }
 
   private static Arguments parseArguments(String[] args) {
-    if (args.length < 4) {
+    if (args.length < 5) {
       logger.error("Wrong number of parameters.");
       printUsageAndExit();
     }
 
-    int nThreads = 1;
+    int nThreads = parseIntArg(args[0], "number of threads");
+    String imageName = args[1];
+    int diskSizeGb = parseIntArg(args[2], "disk size");
+    String directoryToScan = args[3];
+    String manifestFilePath = args[4];
+
+    return new Arguments(nThreads, imageName, diskSizeGb, directoryToScan, manifestFilePath);
+  }
+
+  private static int parseIntArg(String arg, String argName) {
     try {
-      nThreads = Integer.parseInt(args[0]);
-      if (nThreads < 1) {
+      int intVal = Integer.parseInt(arg);
+      if (intVal < 1) {
         throw new NumberFormatException();
       }
+      return intVal;
     } catch (NumberFormatException e) {
-      logger.error("Wrong argument type - number of threads should be a positive integer");
+      logger.error("Wrong argument type - {} should be a positive integer", argName);
       printUsageAndExit();
```

Comment:
I don't know if Java has the concept of a `Nothing` type, but if so, you could possibly `return printUsageAndExit()` and avoid the confusing `return -1`

Reply:
Yeah, that would have been nice, but unfortunately Java doesn't have `Nothing` type.
Alternative way would be to make method return type `Integer` instead of `int` and return `null` in `catch` block. But I'm not sure it would look better then current implementation with `return -1`.